src/js/tech/chromecast.js

```diff
@@ -23,8 +23,19 @@ class Chromecast extends Tech {
         this.apiSession = this.options_.source.apiSession;
         this.receiver = this.apiSession.receiver.friendlyName;
 
-        this.apiMedia.addUpdateListener(::this.onMediaStatusUpdate);
-        this.apiSession.addUpdateListener(::this.onSessionUpdate);
+        let mediaStatusUpdateHandler = ::this.onMediaStatusUpdate;
+        let sessionUpdateHanlder = ::this.onSessionUpdate;
+
+        this.apiMedia.addUpdateListener(mediaStatusUpdateHandler);
+        this.apiSession.addUpdateListener(sessionUpdateHanlder);
+
+        this.on('dispose', () => {
```

Comment:
if `mediaStatusUpdateHandler` and others are available on the instance, you can also just override the `dispose` method and add that in there (remembering to call super) instead of listening to `dispose`